src/main/java/seedu/duke/controllers/OrderController.java

```diff
@@ -50,17 +50,11 @@ protected boolean optionSwitcher(int choice) throws OperationTerminationExceptio
 
     private void addOrder() throws OperationTerminationException {
         boolean notQuit = true;
-        String userInput = InputParser.getString("Enter dishes you want to order (Q/q to exit): ");
-        if (userInput == "Q" || userInput == "q") {
-            notQuit = false;
-        }
+        String userInput = InputParser.getString("Enter dishes you want to order (enter '-' to exit): ");
         while (notQuit) {
             Object inputObj = userInput;
             int size = orderManager.addDishToOrder(inputObj);
-            userInput = InputParser.getString("You’ve already added " + size + "dish(es), some more: \n");
-            if (userInput == "Q" || userInput == "q") {
-                notQuit = false;
-            }
+            userInput = InputParser.getString("You’ve already added " + size + " dish(es), some more: \n");
         }
```

Comment:
can simply be `while (true)`

Reply:
btw, the gradle tests fail now lol

Reply:
it seems that the constructor is useful

Reply:
doesnt need to be generic tho, can just be `List<Dish> dishList`